webui/src/Main.elm

```diff
@@ -180,11 +184,27 @@ update msg model =
                 ( { model | commentsModel = commentsModel }, Cmd.map CommentsMsg cmd )
 
         SurveyMsg surveyMsg ->
-            let
-                ( surveyModel, cmd ) =
-                    Survey.update surveyMsg model.surveyModel model.authModel
-            in
-                ( { model | surveyModel = surveyModel }, Cmd.map SurveyMsg cmd )
+            case surveyMsg of
+                Survey.SaveCurrentSurvey ->
+                    let
+                        ( surveyModel, cmd ) =
+                            Survey.update surveyMsg model.surveyModel model.authModel
+                        ( surveyResponseModel, respCmd ) =
+                            SurveyResponses.update SurveyResponses.GetResponses model.surveyResponseModel model.authModel
+                        _= 
+                            Debug.log "SavedCurrentSurvey " (toString respCmd)
+                    in
+                        ( { model | surveyModel = surveyModel, surveyResponseModel = surveyResponseModel }
+                        , Cmd.batch [ Cmd.map SurveyResponseMsg respCmd, Cmd.map SurveyMsg cmd ]
```

Comment:
We figured out that this Cmd.batch doesn't guarantee execution order (https://www.reddit.com/r/elm/comments/6as1re/platformcmdbatch_executes_commands_in_reverse/). We need to chain the HTTP requests using Http.toTask and Task.andThen. More details here http://tech.allo-media.net/learning/elm/2018/02/05/chaining-http-requests-in-elm.html and here